CSP/CspInstance.cs

```diff
@@ -72,8 +87,34 @@ public void RemoveColor(Pair pair)
             pair.Variable.avalibleColors.Remove(pair.Color);
         }
         public void RemoveColor(Variable variable, Color color) => RemoveColor(new Pair(variable, color));
-       
 
+        #endregion
+
+        #region variable
+        public void AddVariable(IEnumerable<Color> variableColors)
+        {
+            var v = new Variable(variableColors);
+            Variables.Add(v);
+            foreach (var c in variableColors)
+            {
+                AddColor(new Pair(v, c));
+            }
```

Comment:
Jeśli używasz konstruktora zmiennej z przekazywaną listą kolorów to nie musisz ich już dodawać potem, konstruktor to robi